nixos/modules/services/networking/syncthing.nix

```diff
@@ -616,7 +610,7 @@ in {
         };
       };
       syncthing-init = mkIf (
-        cfg.devices != {} || cfg.folders != {} || cfg.extraOptions != {}
+        cfg.settings.devices != {} || cfg.folders != {} || cfg.extraOptions != {}
```

Comment:
This looks suspicious

Reply:
Indeed this looks bad. What do you think about simplify it to:

```diff
diff --git i/nixos/modules/services/networking/syncthing.nix w/nixos/modules/services/networking/syncthing.nix
index b5ebda6da04..d5531409428 100644
--- i/nixos/modules/services/networking/syncthing.nix
+++ w/nixos/modules/services/networking/syncthing.nix
@@ -610,7 +610,7 @@ in {
         };
       };
       syncthing-init = mkIf (
-        cfg.settings.devices != {} || cfg.folders != {} || cfg.extraOptions != {}
+        cfg.settings != {}
       ) {
         description = "Syncthing configuration updater";
         requisite = [ "syncthing.service" ];
```

Reply:
sure

Reply:
That condition hasn't prevented the `syncthing-init.service` from being generated, and this is not the issue @azahi is experiencing. It is still worth fixing though.